### PR TITLE
[Manual backport 2.x]Updates NOTICE file, adds validation to GitHub CI & Corrects NOTICE f…

### LINKED ISSUE #765: [BUG] Discrepancy between generated NOTICE file and the file in the repository
**Describe the bug**

The [NOTICE](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.0/NOTICE.txt) file that exists in the repository has been manually edited thus far, and it does not match the generated NOTICE file using the [notice generator](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.0/src/dev/notice/generate_notice_from_source.ts).

**Expected behavior**
Use the notice generator to make changes to the NOTICE file instead of manual changes.

### PR DESCRIPTION
**Pull request by kristenTian**
### Description
* Aligns NOTICE with rules outlined in opensearch-project/.github#21.
* Adds NOTICE validation to the build and test workflow.
* Fixes product name discrepancies between repository NOTICE file and the generated NOTICE file for the build.
* Skips template-izing the build and test workflow since that's being worked on in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2991

### Resolved Issue
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/765

### Backport PR
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3051
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3308
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ananzh — APPROVED** [review]
LGTM

File: NOTICE.txt [hunk 1]
```diff
@@ -1,7 +1,15 @@
-OpenSearch
-Copyright 2021 OpenSearch Contributors
-This product includes software, including Kibana source code, developed by Elasticsearch (http://www.elastic.co).
-Copyright 2012-2021 Elasticsearch B.V.
+OpenSearch (https://opensearch.org/)
+Copyright OpenSearch Contributors
+
+This product includes software, including Kibana source code,
+developed by Elasticsearch (http://www.elastic.co).
+Copyright 2009-2021 Elasticsearch B.V.
```
Comment (joshuarrrr) [inline NOTICE.txt:6]: OK, I'm still not convinced we have the dates correct - we're going from `2012-2021` to `2009-2021`.  We fixed the end date in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3308/files, but I don't expect either of these years to change.

**Review by joshuarrrr — DISMISSED** [review]
I don't think we intend to change the copyright start date? Blocking until we can confirm.

File: NOTICE.txt (hunk 1 above)
Comment (joshuarrrr) [reply NOTICE.txt:6]: @tmarkley Can you clarify?

Comment (ashwin-pc) [reply NOTICE.txt:6]: The new value is correct. It should not change in theory but the purpose of the date is to indicate the start and end year when the code was first created, and there is definitely an instance that mentions that the code was created in 2009 so updating this is correct.

**Review by joshuarrrr — APPROVED** [review]

**Pull request merged**